pkg/image/registry/image/etcd/etcd.go

```diff
@@ -59,8 +59,27 @@ func NewREST(optsGetter restoptions.Getter) (*REST, error) {
 	}
 
 	if err := restoptions.ApplyOptions(optsGetter, store, prefix); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &REST{store}, nil
+	signatureStore := *store
+	signatureStore.UpdateStrategy = image.SignatureStrategy
+	signatureStore.CreateStrategy = nil
+
+	return &REST{store}, &SignatureREST{store: &signatureStore}, nil
+}
+
+// SignatureREST implements a RESTStorage for signatures subset of images against etcd.
+type SignatureREST struct {
+	store *etcdgeneric.Etcd
+}
+
+// New returns a new image object.
+func (r *SignatureREST) New() runtime.Object {
+	return &api.Image{}
+}
+
+// Update alters the signatures subset of an image object.
+func (r *SignatureREST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
```

Comment:
Obviously, I'd be looking for updates here to match the changes I suggested in the client code.